tests/core/go_library/BUILD.bazel

```diff
@@ -28,8 +28,10 @@ go_library(
     srcs = [
         "inc.go",
         "inc_asmhdr.s",
+        "inc_relative.s",
```

Comment:
This tests including headers from direct dependencies with relative paths, but I think it's also important to test including headers from transitive dependencies. Let's add `tests/core/transitive_headers` with three packages each with a header file (`a.h`, `b.h`, and `c.h`). `a.h` should include `b.h`, and `b.h` should include `c.h`. The package for `a` should not directly depend on `c`, though the transitive header will be needed for compilation to succeed.